src/fork-stream.ts

```diff
@@ -0,0 +1,42 @@
+/**
+ * @license
+ * Copyright (c) 2016 The Polymer Project Authors. All rights reserved.
+ * This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+ * The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+ * The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+ * Code distributed by Google as part of the polymer project is also
+ * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+import {PassThrough, Readable, Transform} from 'stream';
+
+export function forkStream(stream: NodeJS.ReadableStream) {
+  return new ForkedVinylStream(stream);
+}
+
+/**
+ * Forks a stream of Vinyl files, cloning each file before emitting on the fork.
+ */
+export class ForkedVinylStream extends Readable {
+
+  input: NodeJS.ReadableStream;
+
+  constructor(input: NodeJS.ReadableStream) {
+    super({objectMode: true});
+    this.input = input;
```

Comment:
I just don't like more than one way to do things so fundamental... I want to look in one place for the class fields. I could make a local var, but, eh...